File: src/courses/data/catalog/02-dsa/dsa-core/implementations/javascript/heap.js
/**
 * JavaScript Implementation: Heap & Priority Queue
 * Invariant: Insert O(log N), ExtractMin O(log N), Peek O(1)
 */
export class HeapPriorityQueue {
  constructor() {
    this.items = [];
  }

  // TODO: Implement verified invariant
  insert(val) {
    this.items.push(val);
  }

  get size() {
    return this.items.length;
  }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/heap.java
/**
 * Java Implementation: Heap & Priority Queue
 * Invariant: Insert O(log N), ExtractMin O(log N), Peek O(1)
 */
package seed.dsa.core;

import java.util.*;

public class HeapPriorityQueue<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
